### PR TITLE
impl: test sub cmd with support for --manifest-path (#280)

### PR DESCRIPTION
**Pull request by chiefbiiko**
drafts `cargo contract test [FLAGS] [--manifest-path]`

in this first impl stdout is uncolored

let me know if you care much for the colors

### CODE REVIEW

**Review by Robbepop — COMMENTED** [review]
What precisely is the win of having `cargo contract test` over just using `cargo test` as it can be done right now?

**Comment by cmichi** [comment]
@Robbepop At the moment `cargo contract --help` shows the sub-command as:
```
test     Test the smart contract off-chain
```
If one invokes `cargo contract test` currently this results in an `unimplemented!` panic though, which is why people were irritated and started creating issues about it in this repository.

So we could either remove the sub-command or just hand over to `cargo test`. After talking to @ascjones we concluded that it's probably best to just spawn a sub-process with `cargo test` until we have further ideas what to do with the sub-command. 

What's your opinion?

**Comment by Robbepop** [comment]
> @Robbepop At the moment `cargo contract --help` shows the sub-command as:
> 
> ```
> test     Test the smart contract off-chain
> ```
> 
> If one invokes `cargo contract test` currently this results in an `unimplemented!` panic though, which is why people were irritated and started creating issues about it in this repository.
> 
> So we could either remove the sub-command or just hand over to `cargo test`. After talking to @ascjones we concluded that it's probably best to just spawn a sub-process with `cargo test` until we have further ideas what to do with the sub-command (#280).
> 
> What's your opinion?

I currently do not see a major reason but it might help against confusion. So yeah let's do it the proposed way.

File: src/cmd/test.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#[cfg(test)]
+mod tests_ci_only {
+    use crate::{util::tests::with_new_contract_project, Verbosity};
+
+    #[test]
+    fn passing_tests_yield_stdout() {
+        with_new_contract_project(|manifest_path| {
+            let res = super::execute(&manifest_path, Verbosity::Default).expect("test failed");
+
+            assert!(res.stdout.len() > 0);
```
Comment (cmichi) [inline src/cmd/test.rs:83]: Could you add another assertions which checks for something like `test result: ok. 2 passed; 0 failed; 0 ignored`?

This way we can then be sure that there was no other unrelated failure the cause of `stdout.len() > 0`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.13.0] - 2021-05-27
+
+### Added
+- Convenient offchain testing through `cargo contract test`
```
Comment (cmichi) [inline CHANGELOG.md:12]:
```suggestion
- Convenient off-chain testing through `cargo contract test`
```

**Review by cmichi — CHANGES_REQUESTED** [review]
Thanks for the PR, it looks good!

I've got just some minor things. We use the writing `off-chain` everywhere, hence the suggestion of keeping it consistent.

**Comment by chiefbiiko** [comment]
thanks for the review
i have included your suggestions now

**Review by cmichi — APPROVED** [review]
Thanks for the PR!

**Pull request merged**